src/ruby/puppetserver-lib/puppet/server/execution.rb

```diff
@@ -16,6 +16,8 @@ def self.initialize_execution_stub
   end
 
   private
+  SHELL_CHARACTERS = "*?{}[]<>()~&|\\$;'`\"\n#=".chars.to_a
```

Comment:
The `=` here differs slightly from MRI ruby. MRI ruby does special handling to check for `=` *before* the binary (for environment variables), and only invokes the shell in that case. This means we will run using `sh` for any command with `=` before or after, while MRI will only invoke `sh` with `=` before the command.

Reply:
Why `to_a`? What does `chars` return?

Reply:
in 1.9 it returns an enumerator. in 2.x, an array :(

Reply:
want a comment about that?

Reply:
too late. you get a comment